tests/utils.py

```diff
@@ -1,30 +1,107 @@
+"""
+Utility functions for testing
+"""
+
 import os
+import pathlib
 import platform
 import re
 import subprocess
-import sys
 import tempfile
-from pathlib import Path
-from typing import Iterable, Union, NewType, List
+import unittest
+from urllib.request import urlopen
+import warnings
+import enum
 
-TMPDIR = Path(tempfile.gettempdir())
-DEFAULT_TARGET = TMPDIR / "default.out"
-GEF_PATH = Path(os.getenv("GEF_PATH", "gef.py"))
-STRIP_ANSI_DEFAULT = True
-DEFAULT_CONTEXT = "-code -stack"
+from typing import Iterable, Union, List, Optional
+
+TMPDIR = pathlib.Path(tempfile.gettempdir())
 ARCH = (os.getenv("GEF_CI_ARCH") or platform.machine()).lower()
+BIN_LS = pathlib.Path("/bin/ls")
```

Comment:
this is more of a discussion then a review, but here it comes:

Why do we actually use `/bin/ls` here? Is there any specific reason apart from just being any binary? And if so, why dont we use one of the binaries compiled for the tests or even `/bin/sh`?

One of the reasons I noticed this is because I'm currently trying out NixOS for my main machine and there `/bin/ls` doesnt exist (it has a different path) which is kind of annoying because I have to always do some workarounds to make the tests work for me.  Allthough I'm aware that it doesnt make sense to adjust our tests to such a distro it did raise a valid question about the need to use a hardcoded `/bin/ls` IMO.

What are your opinions on that? @hugsy, @Grazfather, @daniellimws

Reply:
I usually use `/bin/cat` but I agree it would be best to use a binary that we build/place ourselves.

Reply:
@theguy147 yeah no problem at all here. the point was to have a binary that'd work "universally" but I'm not attached to ls 😄 feel free to suggest a better one

Reply:
I would just replace all occurrences of `BIN_LS` with the default target: `_target("default")`. I just did a quick test on your current `better_tests` branch and it seemed to work fine

Reply:
Fixed

Reply:
awesome! now I can test without workarounds :dancers: